feature-pack/src/main/resources/modules/system/layers/base/org/jboss/as/weld/main/module.xml

```diff
@@ -37,6 +37,7 @@
         <module name="org.jboss.weld.core" />
         <module name="org.jboss.weld.spi" />
         <module name="org.jboss.as.weld.spi" />
+        <module name="org.picketbox" />
```

Comment:
why is a dependency to PicketBox required?

Reply:
I think this is because in Weld 3 you have async events and there we need to propagate security context. Hence `WeldSecurityServices` need to operate on classes from picketbox (or elytron, whichever is present) such as [`SecurityContextAssociation`](https://github.com/wildfly/wildfly/pull/10642/files#diff-a03cbe645209f431d4aab4d8f8ecc510R41).

Reply:
i would say elytron is the main thing here, @darranl wdyt?

Reply:
Not having this dependency will blow up with CNFE (on `org.jboss.security.SecurityContextAssociation`) when using async events. The reason is as I described above - we need to propagate security context and for that we need this dependency as the case can be that elytron isn't actually used (in which case there is going to be picketbox, right?).

That's at least from what I understood about elytron/picketbox, feel free to correct me.